### PR TITLE
Add man page

### LINKED ISSUE #279: man page missing
Currently, the `ceph-salt` CLI does not provide documentation in the form of a man page:
```
# man ceph-salt
No manual entry for ceph-salt
```
As man pages are the canonical way of providing documentation for command-line utilities, it would be useful to add one.

### PR DESCRIPTION
**Pull request by ricardoasmarques**
To check the rendered version, just run the following command on project root:

```
$ man ./ceph-salt.8
```

Screenshot from 2020-10-21 14-59-04

Fixes: https://github.com/ceph/ceph-salt/issues/279

Signed-off-by: Ricardo Marques <rimarques@suse.com>

### CODE REVIEW

File: ceph-salt.8 [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+performed
+.RE
+.sp
+\fBexport\fP [\fIoptions\fP]
+.RS 4
+Export configuration
+.sp
+\fB\-p\fP, \fB\-\-pretty\fP
+.RS 4
+Pretty-prints JSON ouput
```
Comment (asettle) [inline ceph-salt.8]: Pretty prints JSON output - unsure if there's a particular reason you're using a dash here?

**Review by asettle — APPROVED** [review]

File: ceph-salt.8 (hunk 1 above)
Comment (smithfarm) [reply ceph-salt.8]: "pretty-prints" is OK in U.S. usage

Comment (asettle) [reply ceph-salt.8]: :woman_shrugging: cool

File: ceph-salt.spec [hunk 2]
```diff
@@ -94,13 +94,18 @@ mkdir -p %{buildroot}%{fdir}/metadata/%{fname}/
...
+cp ceph-salt.8 %{buildroot}%{_mandir}/man8/ceph-salt.8
+
 
 %files
 %license LICENSE
 %doc CHANGELOG.md README.md
 %{python3_sitelib}/ceph_salt*/
 %{_bindir}/%{name}
 %dir %{_datadir}/%{name}
+%{_mandir}/man8/ceph-salt.8*
```
Comment (smithfarm) [inline ceph-salt.spec]: please insert `%doc ` at the beginning of this line

Comment (smithfarm) [reply ceph-salt.spec]: it might also be worth considering to either move the line up so it's adjacent to the other `%doc` line ~, or even merging the two `%doc` lines into one~

Comment (smithfarm) [reply ceph-salt.spec]: On second thought, no, don't merge them into one. This line

```
%doc CHANGELOG.md README.md
```

causes RPM to create a documentation directory for ceph-salt and copy the files into it, whereas this line:

```
%doc %{_mandir}/man8/ceph-salt.8
```

causes RPM to install the file in `%_mandir`.

By the way, looking at e.g. `rpm -ql man-pages`, it seems like the man page should be gzipped (and only the gzipped version should be the RPM)

**Review by smithfarm — CHANGES_REQUESTED** [review]

**Comment by ricardoasmarques** [comment]
@smithfarm done
```
#ceph-salt grep 'doc.*ceph-salt.8' ceph-salt.spec 
%doc %{_mandir}/man8/ceph-salt.8*
```
and
```
master:~ # rpm -ql ceph-salt | grep 'ceph-salt.8'
/usr/share/man/man8/ceph-salt.8.gz
```

Package for testing: https://build.opensuse.org/package/binaries/home:rimarques:branches:filesystems:ceph:octopus/ceph-salt/openSUSE_Leap_15.2

**Review by smithfarm — APPROVED** [review]
LGTM

File: ceph-salt.8 [hunk 3]
```diff
@@ -0,0 +1,213 @@
+\" Man page for ceph-salt
+.TH "ceph-salt" "8" "" "" "ceph-salt"
+.SH "NAME"
+ceph\-salt \- deploy ceph clusters using cephadm
+.SH "SYNOPSIS"
+.sp
+\fBceph-salt\fP [\fI\-\-global\-opts\fP] \fIcommand\fP
```
Comment (LenzGr) [inline ceph-salt.8:7]: The synopsis mentions `[--global-opts]`, but these are not explained/listed anywhere.

Comment (ricardoasmarques) [reply ceph-salt.8:7]: Fixed. I've now added a section for **GLOBAL OPTIONS**.

File: ceph-salt.8 [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+[\fI\-\-command\-opts\fP] [\fIcommand\-arguments\fP]
+.sp
+\fBceph-salt\fP [\fIcommand\fP] \fB\-\-help\fP
+.sp
+\fBceph-salt\fP \fB\-\-help\fP
+.SH "DESCRIPTION"
+.sp
+ceph\-salt is a command\-line tool for deploying ceph clusters, managed by
```
Comment (LenzGr) [inline ceph-salt.8]: I find this sentence somewhat hard to parse. How about this one?

> ceph-salt is  a Salt-based command-line tool for deploying ceph clusters managed by cephadm.

File: ceph-salt.8 [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+.sp
+ceph\-salt provides a number of commands. Each command accepts the options
+listed in the \fBGLOBAL OPTIONS\fP section. These options must be specified
+before the command name. In addition, many commands have specific options,
+which are listed in this section. These command-specific options must be
+specified after the name of the command and before any of the command arguments.
+.sp
+\fBapply\fP [\fIoptions\fP] \fI[minion_id]\fP
+.RS 4
+Apply configuration by running ceph-salt formula in order to provision hosts
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Applies a configuration by running ceph-salt formula in order to provision hosts
```

File: ceph-salt.8 [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+so they can be used as ceph cluster nodes. On first execution, a new ceph
+cluster with a single MON and MGR will be deployed. Apply command is
+idempotent, so it can be executed multiple times.
+.sp
+\fB\-n\fP, \fB\-\-non\-interactive\fP
+.RS 4
+Execute without opening a graphical user interface.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Executes without opening the interactive text-based user interface.
```

File: ceph-salt.8 [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+\fBminion_id\fP
+.RS 4
+The minion that should be configured. If not specified, all ceph-salt minions
+will be configured.
+.RE
+.RE
+.sp
+\fBconfig\fP [\fIpath\fP] \fI[subcommand]\fP
+.RS 4
+Open an interactive shell where you can manipulate ceph-salt's configuration.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Opens an interactive shell where you can manipulate ceph-salt's configuration.
```

File: ceph-salt.8 [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+\fBsubcommand\fP
+.RS 4
+Subcommand to execute on configuration path.
+.RE
+.RE
+.RE
+.sp
+\fBdisengage-safety\fP
+.RS 4
+Disable safety so dangerous operations like purging the whole cluster can be
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Disables safeguards so dangerous operations like purging the whole cluster can be
```

File: ceph-salt.8 [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+performed. Safety will be automatically enabled after 60 seconds.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
performed. Safety will be automatically re-enabled after 60 seconds.
```

File: ceph-salt.8 [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.sp
+\fBexport\fP [\fIoptions\fP]
+.RS 4
+Export configuration.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Exports the configuration.
```

File: ceph-salt.8 [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+.sp
+\fB\-p\fP, \fB\-\-pretty\fP
+.RS 4
+Pretty-prints JSON ouput.
+.RE
+.RE
+.sp
+\fBimport\fP \fIfile\fP
+.RS 4
+Import configuration.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Imports a configuration.
```

File: ceph-salt.8 [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+.sp
+\fBfile\fP
+.RS 4
+The file containing the previously exported configuration.
```
Comment (LenzGr) [inline ceph-salt.8]: In which format?

File: ceph-salt.8 [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.RE
+.sp
+\fBpurge\fP [\fIoptions\fP]
+.RS 4
+Destroy ceph cluster. Before being able to destroy a ceph cluster, the
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Destroys a Ceph cluster. Before being able to destroy a Ceph cluster, the
```

File: ceph-salt.8 [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+\fBdisengage\-safety\fP command must be executed.
+.sp
+\fB\-n\fP, \fB\-\-non\-interactive\fP
+.RS 4
+Execute without opening a graphical user interface.
```
Comment (LenzGr) [inline ceph-salt.8]: See above.

File: ceph-salt.8 [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.sp
+\fB\-\-yes\-i\-really\-really\-mean\-it\fP
+.RS 4
+Confirm I really want to perform this dangerous operation.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Additional required confirmation to perform this dangerous operation.
```

File: ceph-salt.8 [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.RE
+.sp
+\fBreboot\fP [\fIoptions\fP] \fIminion_id\fP
+.RS 4
+Reboot hosts if some processes are using deleted files, which may happen after
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Reboots hosts if some processes are using deleted files, which may happen after
```

File: ceph-salt.8 [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+a system update. If ceph cluster is already deployed, nodes are rebooted
+sequentially in an orchestrated way, otherwise, they are rebooted in parallel.
+.sp
+\fB\-n\fP, \fB\-\-non\-interactive\fP
+.RS 4
+Execute without opening a graphical user interface.
```
Comment (LenzGr) [inline ceph-salt.8]: See above

File: ceph-salt.8 [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+.RS 4
+The minion that should be rebooted. If not specified, all ceph-salt minions
+will be rebooted.
+.RE
+.RE
+.RE
+.sp
+\fBstatus\fP [\fIoptions\fP]
+.RS 4
+Display ceph-salt status, including information about configuration errors and
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Displays ceph-salt status, including information about configuration errors and
```

File: ceph-salt.8 [hunk 19]
```diff
@@ -0,0 +1,230 @@
...
+.sp
+\fB\-n\fP, \fB\-\-no\-color\fP
+.RS 4
+Output without colors.
+.RE
+.RE
+.sp
+\fBstop\fP [\fIoptions\fP]
+.RS 4
+Stop ceph cluster.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Stops the Ceph cluster.
```

File: ceph-salt.8 [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+.sp
+\fB\-n\fP, \fB\-\-non\-interactive\fP
+.RS 4
+Execute without opening a graphical user interface.
```
Comment (LenzGr) [inline ceph-salt.8]: See above

File: ceph-salt.8 [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.sp
+\fB\-\-yes\-i\-really\-really\-mean\-it\fP
+.RS 4
+Confirm I really want to perform this dangerous operation.
```
Comment (LenzGr) [inline ceph-salt.8]: See above

File: ceph-salt.8 [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.RE
+.sp
+\fBupdate\fP [\fIoptions\fP] \fI[minion_id]\fP
+.RS 4
+Update all system packages.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Updates all host operating system packages.
```

File: ceph-salt.8 [hunk 23]
```diff
@@ -0,0 +1,230 @@
...
+.sp
+\fB\-n\fP, \fB\-\-non\-interactive\fP
+.RS 4
+Execute without opening a graphical user interface.
```
Comment (LenzGr) [inline ceph-salt.8]: See above.

File: ceph-salt.8 [hunk 24]
```diff
@@ -0,0 +1,230 @@
...
+.SH "GLOBAL OPTIONS"
+.sp
+\fB\-l\fP, \fB\-\-log\-level\fP \fI[info|error|debug|silent]\fP
+.RS 4
+Specify the log level.
+.RE
+.sp
+\fB\-\-log\-file\fP \fIfile\fP
+.RS 4
+Specify the log file.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Specify the log file name.
```

File: ceph-salt.8 [hunk 25]
```diff
@@ -0,0 +1,230 @@
...
+.SH "EXAMPLES"
+.sp
+\fBDay 1 - deploy a new ceph cluster\fP
+.RS 4
+.sp
+$ \fBceph\-salt config\fP
+.RS 4
+Open the interactive config shell that will be used to configure cluster.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Opens the interactive configuration shell that will be used to configure cluster.
```

File: ceph-salt.8 [hunk 26]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.sp
+$ \fBceph\-salt status\fP
+.RS 4
+Check if provided configuration is valid.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Checks if the provided configuration is valid.
```

File: ceph-salt.8 [hunk 27]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.sp
+$ \fBceph\-salt update --reboot\fP
+.RS 4
+Guarantee that cluster is updated with latest patches.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Ensures that the cluster is updated with latest patches.
```

File: ceph-salt.8 [hunk 28]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.sp
+$ \fBceph\-salt apply\fP
+.RS 4
+Deploy ceph cluster based on provided configuration.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Deploys a Ceph cluster based on the provided configuration.
```

File: ceph-salt.8 [hunk 29]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.RE
+.sp
+\fBDay 2 - add a new host to an existing ceph cluster\fP
+.RS 4
+.sp
+$ \fBceph\-salt config /ceph_cluster/minions add <new_minion_id>\fP
+.RS 4
+.sp
+Add new minion to the configuration, so it will be managed by ceph-salt.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Adds a new minion to the configuration, so it will be managed by ceph-salt.
```

File: ceph-salt.8 [hunk 30]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.sp
+$ \fBceph\-salt config /ceph_cluster/roles/cephadm add <new_minion_id>\fP
+.RS 4
+.sp
+Assign cephadm role to the new minion, so it will be added to the ceph cluster.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Assigns the cephadm role to the new minion, so it will be added to the Ceph cluster.
```

File: ceph-salt.8 [hunk 31]
```diff
@@ -0,0 +1,230 @@
...
+.RE
+.sp
+$ \fBceph\-salt apply <new_minion_id>\fP
+.RS 4
+Apply configuration on the new minion.
```
Comment (LenzGr) [inline ceph-salt.8]:
```suggestion
Applies configuration on the new minion.
```

**Comment by LenzGr** [comment]
Looks good, please consider my enhancement suggestions. I also suggest to write "Ceph" (with a capital "C") consistently in the document. Thanks!

**Comment by ricardoasmarques** [comment]
@LenzGr Done. Thanks for the detailed review.

**Review by LenzGr — APPROVED** [review]
LGTM! Thanks a lot for incorporating my suggestions! LGTM.
